### PR TITLE
added config service

### PR DESCRIPTION
**Pull request by EdgarasDamkus**

### CODE REVIEW

File: media_skyfish.services.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+services:
+  media_skyfish.configservice:
+    class: Drupal\media_skyfish\ConfigService
+    arguments: ['@http_client', '@config.factory']
```
Comment (irmantasp) [inline media_skyfish.services.yml]: User service should be added here too.

File: src/ConfigService.php [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+   */
+  protected $password;
+
+  /**
+   * ConfigService constructor.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   * @param \GuzzleHttp\Client $client
+   */
+  public function __construct(ConfigFactoryInterface $configFactory, Client $client) {
```
Comment (irmantasp) [inline src/ConfigService.php]: in D8, variables are not camel case. you should use underscore. e.g. $config_factory. class variable, can be camel case.

File: src/ConfigService.php [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    $this->client = $client;
+    $this->config = $configFactory->get('media_skyfish.adminconfig');
```
Comment (irmantasp) [inline src/ConfigService.php]: $this->config = $config_factory->get('media_skyfish.adminconfig');

File: src/ConfigService.php [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    $this->user = \Drupal::entityTypeManager()->getStorage('user')->load(\Drupal::currentUser()->id());
```
Comment (irmantasp) [inline src/ConfigService.php]: This could be loaded as service.

File: src/ConfigService.php [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    $this->initialize();
+  }
+
+  private function initialize(){
```
Comment (irmantasp) [inline src/ConfigService.php]: Initialize needs description.

**Review by irmantasp — CHANGES_REQUESTED** [review]

File: src/ApiService.php [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+namespace Drupal\media_skyfish;
+
+use GuzzleHttp\Client;
+
+/**
+ * Class ApiService.
+ */
+class ApiService {
+
+  const MEDIA_SKYFISH_API_BASE_URL = 'https://api.colourbox.com';
```
Comment (mindaugas-laurinaitis) [inline src/ApiService.php]: visibility and phpdoc for cons should be provided

File: src/ApiService.php [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+   * Http client used for connection.
+   *
+   * @var \GuzzleHttp\Client
+   */
+  protected $client;
+
+  /**
+   * ApiService constructor.
+   *
+   * @param \Drupal\media_skyfish\ConfigService $config_service
```
Comment (mindaugas-laurinaitis) [inline src/ApiService.php]: Missing comments of parameters

File: src/ApiService.php [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+    return 'CBX-SIMPLE-TOKEN Token=' . $token;
+  }
+
+  /**
+   * Make request to Skyfish with uri.
+   *
+   * @param $uri
+   *
+   * @return mixed
```
Comment (mindaugas-laurinaitis) [inline src/ApiService.php]: Mixed should not be used. Also almost all phpdocs is missing some information, like parameter comment

**Pull request merged**